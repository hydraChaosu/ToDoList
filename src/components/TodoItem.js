import React from "react"
import styled, { css } from "styled-components"

const Button = styled.button`
  padding: 15px 10px;
  min-width: 100px;
  margin: 5px;
  transition: 0.3s all;
  :hover {
    box-shadow: 0 0 10px yellow;
  }
  ${props =>
    props.com &&
    css`
      background: green;
      color: white;
    `}
  ${props =>
    props.rem &&
    css`
      background: red;
      color: white;
    `}
`
const List = styled.li`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  padding: 0 5px;
  text-decoration: none;
  list-style: none;
  @media (min-width: 768px) {
    padding: 0;
  }
`

const Content = styled.p`
  flex-grow: 1;
`

const TodoItem = ({ content, completed, complete, remove, id }) => {
  return (
    <List>
      <Content>{content}</Content>
      <Button com onClick={() => complete(id)}>
        {completed ? "Uncomplete" : "Complete"}
      </Button>
      <Button rem onClick={() => remove(id)}>
        Remove
      </Button>
    </List>
  )
}

export default TodoItem
