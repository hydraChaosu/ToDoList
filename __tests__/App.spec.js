import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import { shallow } from "enzyme"
import ToDo from "../src/pages/index"
import TodoItem from "../src/components/TodoItem"

import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

configure({ adapter: new Adapter() })

describe("<ToDo/>", () =>
  it("renders without crashing", () => {
    const App = renderer.create(<ToDo />).toJSON()
    expect(App).toMatchSnapshot()
  }))

describe(
  "<TodoItem/>",
  () =>
    it("renders without crashing", () => {
      const Item = shallow(<TodoItem />)
      expect(Item).toMatchSnapshot()
    }),

  it("renders 2 button in item", () => {
    const Item = shallow(<TodoItem />)
    expect(Item.find("button").length).toEqual(2)
  })
)
