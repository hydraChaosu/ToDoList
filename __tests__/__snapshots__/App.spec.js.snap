// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDo/> renders without crashing 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>,
  <p>
    Uncompleted tasks
  </p>,
  <li>
    <p>
      Eat rabarbar
    </p>
    <button
      onClick={[Function]}
    >
      Complete
    </button>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </li>,
  <li>
    <p>
      Eat potato
    </p>
    <button
      onClick={[Function]}
    >
      Complete
    </button>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </li>,
  <li>
    <p>
      Eat meat
    </p>
    <button
      onClick={[Function]}
    >
      Complete
    </button>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </li>,
  <p>
    Completed tasks
  </p>,
  <li>
    <p>
      Eat rice
    </p>
    <button
      onClick={[Function]}
    >
      Uncomplete
    </button>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </li>,
  <li>
    <p>
      Eat cat and dog
    </p>
    <button
      onClick={[Function]}
    >
      Uncomplete
    </button>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </li>,
]
`;

exports[`<TodoItem/> renders without crashing 1`] = `ShallowWrapper {}`;
